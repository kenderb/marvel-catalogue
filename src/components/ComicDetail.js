import React from 'react';

const ComicDetail = () => (
  <div>
    comic

    detail
  </div>
);

export default ComicDetail;
