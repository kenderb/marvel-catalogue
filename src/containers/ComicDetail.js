/* eslint-disable react/no-danger */
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchComicDetail } from '../actions';
import { DetailWrapper } from '../components/styled/lib';

const ComicDetail = ({ details, match, fetchComicDetail }) => {
  const { id } = match.params;
  useEffect(() => {
    fetchComicDetail(id);
  }, [fetchComicDetail]);

  if (details.id) {
    const {
      title, pageCount, thumbnail, description,
      stories, characters,
    } = details;

    const createMarkup = () => (
      { __html: description }
    );

    const { items } = stories;
    const { items: characterList } = characters;
    const thumbnailSplit = thumbnail.path.split('//');
    thumbnailSplit[0] = 'https://';

    return (
      <DetailWrapper>
        <div>
          <h1>
            { title }
          </h1>
          <img src={`${thumbnailSplit.join('')}.${thumbnail.extension}`} alt="" srcSet="" />
        </div>
        <ul>
          <li>
            <b>Number of pages: </b>
            {' '}
            { pageCount }
          </li>
          <li>
            <b>Description: </b>
            <div dangerouslySetInnerHTML={createMarkup()} />
          </li>
          <li>
            <b>
              Characters:
            </b>
            {' '}
            {characterList ? characterList.map(item => <p key={item.name}>{item.name}</p>) : ''}
          </li>
          <li>
            <b>
              Stories:
            </b>
            {' '}
            {items ? items.map(item => <p key={item.name}>{item.name}</p>) : ''}
          </li>
        </ul>
      </DetailWrapper>
    );
  }
  return <div>Loading..</div>;
};

ComicDetail.propTypes = {
  match: PropTypes.instanceOf(Object).isRequired,
  details: PropTypes.instanceOf(Object).isRequired,
  fetchComicDetail: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({ details: state.details });
export default connect(mapStateToProps, { fetchComicDetail })(ComicDetail);
