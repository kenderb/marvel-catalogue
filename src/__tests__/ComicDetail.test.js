import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import combineReducers from '../reducers';
import ComicDetail from '../containers/ComicDetail';

test('ComicDetail renders correctly', () => {
  const mockObj = { params: 'some params' };
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <ComicDetail match={mockObj} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
