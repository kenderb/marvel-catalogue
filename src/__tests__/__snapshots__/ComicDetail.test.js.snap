// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicDetail renders correctly 1`] = `
<div>
  Loading..
</div>
`;
